applications/crossbar/src/cb_apps_util.erl

```diff
@@ -0,0 +1,184 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2012-2015, 2600Hz, INC
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%% Peter Defebvre
+%%%-------------------------------------------------------------------
+-module(cb_apps_util).
+
+-export([allowed_apps/1]).
+-export([allowed_app/2]).
+
+-include("crossbar.hrl").
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Get allowed apps fron service plans
+%% @end
+%%--------------------------------------------------------------------
+-spec allowed_apps(ne_binary()) -> wh_json:objects().
+
+allowed_apps(AccountId) ->
+    ServicePlan = wh_services:service_plan_json(AccountId),
+    case wh_json:get_value(<<"ui_apps">>, ServicePlan) of
+        'undefined' ->
+            load_default_apps();
+        UIApps ->
+            maybe_load_all_apps(UIApps)
+    end.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Get allowed apps fron service plans
+%% @end
+%%--------------------------------------------------------------------
+-spec allowed_app(ne_binary(), ne_binary()) -> api_object().
+allowed_app(AccountId, AppId) ->
+    Apps = allowed_apps(AccountId),
+    filter_apps(Apps, AppId, 'undefined').
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec filter_apps(wh_json:objects(), ne_binary(), 'undefined' | wh_json:object()) -> 'undefined' | wh_json:object().
```

Comment:
Also, I don't think you need a fold-style accumulator, no?

```erlang
filter_apps([], _) -> 'undefined';
filter_apps([App|Apps], AppId) ->
    case wh_json:get_value(<<"_id">>, App) of
        AppId -> App;
        _Id -> filter_apps(Apps, AppId)
end.
```

Or am I missing something?